File: plataforma-academica-spring/src/main/java/com/plataforma_academica/plataforma/service/AmizadeService.java
package com.plataforma_academica.plataforma.service;

import java.util.UUID;

import com.plataforma_academica.plataforma.dto.AmizadeDTO;
import com.plataforma_academica.plataforma.model.Amizade;

import java.util.List;

public interface AmizadeService {
    Amizade enviarSolicitacao(AmizadeDTO amizadeDTO);
    Amizade responderSolicitacao(UUID amizadeId, String acao); // "aceitar" ou "recusar"
    void removerAmizade(UUID amizadeId);
    List<Amizade> listarSolicitacoesPendentes(UUID usuarioId);
    List<Amizade> listarAmigos(UUID usuarioId);
}

File: plataforma-academica-angular/src/app/services/amizade.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Amizade } from '../models/amizade.model';

@Injectable({
  providedIn: 'root'
})
export class AmizadeService {
  private apiUrl = 'http://localhost:8090/api/amizades';

  constructor(private http: HttpClient) { }

  enviarSolicitacao(solicitanteId: string, destinatarioId: string): Observable<Amizade> {
    const payload = { solicitanteId, destinatarioId };
    console.log('AmizadeService - Enviando:', payload);
    return this.http.post<Amizade>(this.apiUrl, payload);
  }

  responderSolicitacao(id: string, acao: 'aceitar' | 'recusar'): Observable<Amizade> {
    return this.http.patch<Amizade>(`${this.apiUrl}/${id}/resposta`, null, { params: { acao } });
  }

  aceitarSolicitacao(id: string): Observable<Amizade> {
    return this.responderSolicitacao(id, 'aceitar');
  }

  recusarSolicitacao(id: string): Observable<Amizade> {
    return this.responderSolicitacao(id, 'recusar');
  }

  removerAmizade(id: string): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }

  listarPendentes(usuarioId: string): Observable<Amizade[]> {
    return this.http.get<Amizade[]>(`${this.apiUrl}/pendentes/${usuarioId}`);
  }

  listarAmigos(usuarioId: string): Observable<Amizade[]> {
    return this.http.get<Amizade[]>(`${this.apiUrl}/amigos/${usuarioId}`);
  }
}
